resource/ui/ts/models/proto.ts

```diff
@@ -113,6 +155,32 @@ module Models {
             available_range_count: number;
         }
 
+        /**
+         * Status is the common interface shared by NodeStatus and StoreStatus.
+         */
+        export interface Status{
```

Comment:
There is no need, you can just assign a NodeStatus to a Status variable and it works.  It's duck typed.